awx/api/serializers.py

```diff
@@ -1756,6 +1745,14 @@ def to_representation(self, obj):
             ret['last_job_host_summary'] = None
         return ret
 
+    def get_has_active_failures(self, obj):
+        return bool(
+            obj.last_job_host_summary and obj.last_job_host_summary.failed
+        )
+
+    def get_has_inventory_sources(self, obj):
+        return obj.inventory_sources.exists()
```

Comment:
This will probably incur an additional query. However, I see it as a valid candidate for `prefetch_related`, because there will be a large amount of duplication in many cases.

Reply:
Yep, I checked, and this *does* add more queries, so I added it to `prefetch_related` and watched the query count go down.

Reply:
Thanks. That might exhaust the actionable comments I had. I will go back over the diff one last time.

Reply:
https://github.com/ansible/awx/pull/5448/files#diff-c14c9d695bfbeb70889f2b42b18ed42dR910